drivers/timer/nrf_rtc_timer.c

```diff
@@ -222,13 +225,56 @@ void z_nrf_rtc_timer_compare_set(uint32_t chan, uint32_t cc_value,
 	z_nrf_rtc_timer_compare_int_unlock(chan, key);
 }
 
+uint32_t z_timer_cycle_get_32(void)
+{
+	/* Lockless algorithm for reading cycles assumes that there is an
```

Comment:
Is it really necessary for the anchor logic to be so complex?

As far as I can tell, the longest delay that's possible is half the 24-bit span, so unless the RTC interrupt is delayed for over 256 s we know that it will be invoked before the 24-bit counter wraps at 512 s.

We already add the 24-bit delta to `last_count` in the ISR.  Why can't we just use `last_count` as the anchor?

Reply:
yeah, it was a bit too much. I've updated `sys_clock_cycle_get_32` to simply attempt to calculate current value based on `last_count` and repeat until `last count` is unchanged during the process. It is slightly faster but mainly it is lockless.